src/components/Swap/Swap.tsx

```diff
@@ -283,14 +284,19 @@ export const Swap: React.FC<ISwap> = ({
     ) {
       return 'Insufficient balance'
     }
-    if (!simulateResult.simulateSuccess) {
-      return 'Exceed single swap limit (split transaction into several)'
-    }
 
     if (printBNtoBN(amountFrom, tokens[tokenFromIndex].decimals).eqn(0)) {
       return 'Insufficient volume'
     }
-
+    if (simulateResult.error === 'Error: Too large amount') {
+      return 'Exceed single swap limit (split transaction into several)'
+    }
+    if (
+      simulateResult.error === 'Error: At the end of price range' ||
+      simulateResult.error === 'Error: Price would cross swap limit'
```

Comment:
this could be a separate message
It is thrown when price after the swap would be greater than selected slippage. So something lik 'Slippage greater than selected'